src/client.js

```diff
@@ -287,7 +287,15 @@ export function MatrixClient(opts) {
                 _updatePendingEventStatus(room, eventToSend,
                                           EventStatus.SENDING);
             }
-            return _sendEventHttpRequest(self, eventToSend);
+            const promise = _sendEventHttpRequest(self, eventToSend);
+            if (room) {
+                // ensure we update pending event before the next scheduler run so that any listeners to event id
+                // updates on the synchronous event emitter get a chance to run first.
+                promise.then(res => {
```

Comment:
should this be returned, so anyone waiting on the promise also waits on this? if not, probably best to leave a comment that this is deliberate